backend/benefit/applications/services/ahjo_integration.py

```diff
@@ -358,44 +364,132 @@ def export_application_batch(batch) -> bytes:
+    """Get the access token from Ahjo Service."""
     try:
         ahjo_auth_code = AhjoSetting.objects.get(name="ahjo_code").data
         LOGGER.info(f"Retrieved auth code: {ahjo_auth_code}")
+        connector = AhjoConnector(requests)
+
+        if not connector.is_configured():
+            LOGGER.warning("AHJO connector is not configured")
+            return
+        return connector.get_access_token(ahjo_auth_code["code"])
     except ObjectDoesNotExist:
         LOGGER.error(
             "Error: Ahjo auth code not found in database. Please set the 'ahjo_code' setting."
         )
         return
-
-    connector = AhjoConnector(requests)
-
-    if not connector.is_configured():
-        LOGGER.warning("AHJO connector is not configured")
-        return
-    try:
-        ahjo_token = connector.get_access_token(ahjo_auth_code["code"])
     except Exception as e:
         LOGGER.warning(f"Error retrieving access token: {e}")
         return
-    headers = {
-        "Authorization": f"Bearer {ahjo_token.access_token}",
+
+
+def prepare_headers(access_token: str, application_uuid: uuid) -> dict:
+    """Prepare the headers for the Ahjo request."""
+    url = reverse("ahjo_callback_url", kwargs={"uuid": str(application_uuid)})
+
+    return {
+        "Authorization": f"Bearer {access_token}",
+        "Accept": "application/hal+json",
+        "X-CallbackURL": f"{settings.API_BASE_URL}{url}",
     }
-    print(headers)
+
+
+def get_application(id: uuid) -> Optional[Application]:
+    """Get the first accepted application."""
+    application = (
+        Application.objects.filter(pk=id, status=ApplicationStatus.ACCEPTED)
+        .prefetch_related("attachments", "calculation", "company")
+        .first()
+    )
+    if not application:
+        LOGGER.info("No applications found for Ahjo request.")
+    # Check that the handler has an ad_username set, if not, log an error and return None
+    if not application.calculation.handler.ad_username:
+        LOGGER.error("No ad_username set for the handler for Ahjo request.")
+        return None
+    return application
+
+
+def create_status_for_application(application: Application):
+    """Create a new AhjoStatus for the application."""
+    AhjoStatus.objects.create(
+        application=application, status=AhjoStatusEnum.REQUEST_TO_OPEN_CASE_SENT
+    )
+
+
+def do_ahjo_request_with_form_data(
+    url: str,
+    headers: dict,
+    data: dict,
+    application: Application,
+):
+    json_data = json.dumps(data)
+    form_data, content_type = encode_multipart_formdata({"case": json_data})
+
+    headers["Content-Type"] = content_type
+
+    try:
+        request = urllib.request.Request(
+            f"{url}/cases",
+            method="POST",
+            headers=headers,
+            data=form_data.encode("utf-8"),
+        )
+
+        with urllib.request.urlopen(request) as response:
+            response_data = response.read()
+            print(response.status)
```

Comment:
Debug prints I guess?

Reply:
Will remove with the other form-data related code.